### PR TITLE
Feat/swarinfo 154 decode from stream

### PR DESCRIPTION
**Pull request by xgroleau**
Added fixed buffer for now, will migrate to callbacks eventually

### CODE REVIEW

File: hivemind-host/proto/function-call.proto [hunk 1]
```diff
@@ -8,8 +8,8 @@ message FunctionArgument {
 }
 
 message FunctionCallRequest {
-    string functionName = 1;
-    repeated FunctionArgument arguments = 2;
+    string function_name = 1 ;
```
Comment (t-constantin) [inline hivemind-host/proto/function-call.proto:11]: undesired space between "1" and ";" (same for next line)

**Review by t-constantin — APPROVED** [review]
Code  is very clean and well written. I had to search to find something to say.
Note that I am not used too much to cmake, which makes my review not so useful...

**Review by stlaperriere — APPROVED** [review]
LGTM :+1:

File: hivemind-host/CMakeLists.txt [hunk 2]
```diff
@@ -2,9 +2,12 @@ set(LIB_NAME "swarmus-pheromones-hivemind-host")
 set(LIB_ALIAS "SwarmUS::Pheromones::HiveMind::Host")
 
 nanopb_generate_cpp(PROTO_SRCS PROTO_HDRS
-  generic.proto
-  function-call.proto
-  message.proto
+    RELPATH
```
Comment (g-lesssard) [inline hivemind-host/CMakeLists.txt:5]: Question: what is the use of the RELPATH keyword here?

**Review by g-lesssard — APPROVED** [review]

File: hivemind-host/CMakeLists.txt (hunk 2 above)
Comment (xgroleau) [reply hivemind-host/CMakeLists.txt:5]: Allows nanopb to compile relative to this path and not the current one where the call is done. So your imports are relative to where the protofiles are insteand of "proto/file.proto"

**Pull request merged**